### ISSUE TITLE
'Red stone

### ISSUE THREAD

**Issue by Funguss** [CleverRaven/Cataclysm-DDA#37969]
# Describe the bug

# Steps To Reproduce

1. Make a player with a gun, magazine, and some ammo.
2. Try to aim at a wall.
3. There's no way to increase aim level.

# Expected behavior

Ideally also describe *why* you expect it to happen.

# Screenshots

**If applicable**, add screenshots to help explain your problem.

# Versions and configuration

# Additional context

**Comment by Funguss** [comment]
Sooo... After going through and learning about JSON, uploading, fixing the stone and checking out everything is in order, I decided to check out green and blue stone, which were also missing key flags. In doing so I discovered that they're the blocks from the strange temple, and I'm totally uncertain about touching these blocks at all now. Dammit.

**Comment by eilaattwood** [comment]
Well, you can test how strange temple puzzle works after editing flags.

**Comment by Funguss** [comment]
Yeah, just that I've never made my way down one, I didn't really want to spoil it for myself... But yes, I assumed it was something like that. Though by reading up on the tags it seems that NOITEM shouldn't destroy items, though I'm not sure if it would really behave that way.

**Comment by eilaattwood** [comment]
Actually I'm 90% sure someone who designed strange temple and defining colored stone blocks just forgot to add a flag.

**Comment by Funguss** [comment]
Well, it was missing a host of flags, except MINEABLE. They're missing "NOITEM", "SUPPORTS_ROOF", "WALL", and "BLOCK_WIND", along with a roof designation. "roof": "t_rock_floor" seemed like it would do the trick. I've modified my files, and I'm unlikely to tackle a strange temple in the near or distant future, so it's not something I'm prepared to test out unfortunately. I thought this would be a straightforward fix...

**Comment by ZhilkinSerg** [comment]
> That said, my lighthouse is suddenly surrounded by a mysterious darkness one tile thick, so I've probably screwed something up.

Must be shadow.

**Comment by eilaattwood** [comment]
> roof designation

Oh, this definitely should be tested in the strange temple. As the puzzle makes colored stone tiles to disappear and reappear, disappearing roof can potentially lead to interesting results.

**Comment by Funguss** [comment]
Yeah, it must be, but there's nothing to cast a shadow that I can see.

**Comment by ZhilkinSerg** [comment]
> 
> 
> Yeah, it must be, but there's nothing to cast a shadow that I can see.

A roof (probably autogenerated).

**Comment by Funguss** [comment]
Ok, I know that's the regular cause, but in this case it isn't. Some screenshots may help...
![Z0 - Normal](https://user-images.githubusercontent.com/57632331/74412010-6d76f580-4e34-11ea-9df1-d2b2a4ee4848.png)

This is how it looks from the player perspective at Z0. Next will be in map editor, each increasing in Z level, until there is light. Note the light_at figures.

![Z0](https://user-images.githubusercontent.com/57632331/74412226-dbbbb800-4e34-11ea-9ad4-4bfcb3795e60.png)
![Z1](https://user-images.githubusercontent.com/57632331/74412256-e8401080-4e34-11ea-8f53-2c5f5ad3aab0.png)
![Z2](https://user-images.githubusercontent.com/57632331/74412259-e8d8a700-4e34-11ea-839e-44183bb5461d.png)
![Z3](https://user-images.githubusercontent.com/57632331/74412260-e9713d80-4e34-11ea-8221-b5b2c0a2749d.png)

I absolutely cannot explain this. There isn't a roof tile anywhere to be seen and it's not marked as interior.

**Comment by curstwist** [comment]
Hey,  as far as I know, the color stones were always intended to be underground, and should be more like t_rock.

I'm all for making a new line of colored stone walls that are human-made and using those in the lighthouse or other locations.

I've only done the strange temple once or twice but I don't think you needed to put anything on those walls, it's probably an oversight since it's one of the oldest dungeons in the game and those walls are almost never used.  

There are a number of places suffering from the dark overhang and I can't really figure out why.  Others include: Zoo, Migo tower/camp.

udpate: reading about t_rock_floor.  This was always used underground until recently, so maybe it was designed that way because of underground light issues?  Which should've been fixed when Kevin did his big raycasting/shadowcasting work earlier this year.

**Comment by Funguss** [comment]
Nice one on pointing out those other instances! I'm not sure precisely what is going on but I have an inkling it's some kind of roof or cave generation code. Some roof types are creating overhangs when they aren't part of a room which set all the tiles one Z level below in a circle one tile radius to `outside:0`. It seems that any walls with roofs that are not connected to a full roof will do this, but I've not got the complete picture yet.

For example, switching out `t_rock_floor` with `roof_flat` fixes up the lighthouse tiles but has no impact on the mi-go scout towers. The issue in zoos (one at least, there may be others) is the use of `t_open_air_rooved` in the templates for use above the clay walls. This happens in the lighthouse, too, and in both cases do not behave as intended. The floor of that Z level is set to open air while a flat roof is generated above. So, Z0 has brick wall. Z1 has open air. Z2 has flat roof. I imagine the authors intended these to be flat roof all along and some change occurred to the tile or it was a misunderstanding. Either way, these floating flat roofs are causing false interiors below them, creating rays of impenetrable (by sunlight). darkness.

Rather than it being some kind of cave code, it might be a bug in the lighting code. I think I'm at my limit of what I can figure out just using the debug menu.
